core/src/main/java/org/apache/iceberg/TableUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public class TableUtil {
+  private TableUtil() {}
+
+  /**
+   * Returns the format version of the given table. Note that subclasses of {@link
+   * BaseMetadataTable} do not have a format version. The only exception to this is {@link
+   * PositionDeletesTable}, where the format version is fetched from the underlying table.
+   */
+  public static int formatVersion(Table table) {
+    Preconditions.checkArgument(null != table, "Invalid table: null");
+
+    if (table instanceof SerializableTable) {
+      return formatVersion(((SerializableTable) table).underlyingTable());
+    } else if (table instanceof PositionDeletesTable) {
```

Comment:
I feel it is a bit weird to make this exception for `PositionDeletesTable` in core, given that we add it exclusively because of how the maintenance is written in Spark. I'd either return the underlying table format version for all metadata tables (which is pretty questionable as they don't really have a format version) or unwrap the underlying table in Spark. For instance, we already have [a class check](https://github.com/apache/iceberg/blob/f3560871564c95a0e7b2bff3ca6ecb2e08726d01/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java#L289) in `SparkTable`. We can do similar in the maintenance procedure.

Reply:
@RussellSpitzer had more or less the same feedback on this, so let's remove this handling for `PositionDeletesTable` from core and do it in the maintenance procedure

Reply:
Me and Anton are on the same psychic link